### ISSUE TITLE
package.json wrong directory?

### ISSUE THREAD

**Issue by adamklein** [jepiqueau/vue-sqlite-hook#5]
When installing the vue-sqlite-hook via npm (technically I'm using pnpm), I get the following entries in `dist/package.json`:

```
  ...
  "main": "dist/esm/index.js",
  "typings": "dist/esm/index.d.ts",
  ...
```

however, both index.js and index.d.ts are in the same directory as package.json (ie, `./dist`) and not in a `./dist/esm` subdirectory.

As a result, my Vue 3 project using the Vite build tool gives me the following error:

```
[vite:resolve] Failed to resolve entry for package "[...]/vue-sqlite-hook/dist". The package may have incorrect main/module/exports specified in its package.json
```

This error goes away when I change "main" and "typings" entries in `package.json` to get rid of "dist/esm" path prefix (to do so, I'm installing a local checkout of the repository into node modules), ie:

```
  ...
  "main": "index.js",
  "typings": "index.d.ts",
  ...
```

Any ideas where things might be going wrong?

Thx!!

**Comment by jepiqueau** [comment]
@adamklein it is a problem with Vite which seems to require static module and not lazy loading esm module. It was raise by someone to the ionic team. i do not know Vite so if you can share a small app showing the issue, i could look at it and try to figure out a solution.

**Comment by jepiqueau** [comment]
@adamklein this has been fixed in release 2.1.4 of `vue-sqlite-hook` . A demo app has been published [https://github.com/jepiqueau/vuevite-app](https://github.com/jepiqueau/vuevite-app). Thanks for mentioning this issue
